### ISSUE TITLE
QueryID needed from query data for Algolia click analytics

### ISSUE THREAD

**Issue by harrison-symes** [algolia/docsearch#555]
**Is your issue related to your own DocSearch implementation?**

For custom implementation with Algolia Click Analytics (https://www.algolia.com/doc/guides/insights-and-analytics/analytics-overview/#analytics---implementation-overview) the QueryID is needed from the query request data response. 

Within `src/lib/DocSearch.js`, the `getAutocompleteSource` function performs a search query from the Algolia client. 

``` js 
 this.client
        .search([
          {
            indexName: this.indexName,
            query,
            params: this.algoliaOptions,
          },
        ])
        .then(data => {
          // This data variable here will contain the needed information
          let hits = data.results[0].hits;
          if (transformData) {
            hits = transformData(hits) || hits;
          }
          callback(DocSearch.formatHits(hits));
        });
```

The QueryId can be found within the response data, but the Docsearch API does not currently allow the developer to access this full response object, only the `data.results[0].hits` through various means.

I propose that an optional `queryDataCallback` argument for the Docsearch instance be implemented that will call back this full response `data` object so a developer may observe and interact with all response information, rather than just `hits`

**Comment by harrison-symes** [comment]
https://github.com/algolia/docsearch/pull/556

**Comment by Haroenv** [comment]
closed by #556
